lib/App/Config/Chronicle.pm

```diff
@@ -393,6 +409,396 @@ sub _add_dynamic_setting_info {
+
+Example:
+    get(['key1']);
+Returns:
+    {'key1' => 'value1', '_global_revision' => '<number>'}
+
+Example:
+    get('key1');
+Returns:
+    'value1'
+
+=cut
+
+sub get {
+    my ($self, $keys) = @_;
+    my $single_get;
+
+    unless (ref $keys) {
+        $keys       = [$keys];
+        $single_get = 1;
+    }
+
+    $self->_key_exists($_) or die "Cannot get with key: $_ | Key must be defined" foreach @$keys;
+
+    push @$keys, '_global_rev' unless $single_get;
+
+    my @result_objs = $self->_retrieve_objects($keys);
+
+    if ($single_get) {
+        return $result_objs[0] ? $result_objs[0]->{data} : $self->get_default($keys->[0]);
+    } else {
+        return {map { $keys->[$_] => $result_objs[$_] ? $result_objs[$_]->{data} : $self->get_default($keys->[$_]) } (0 .. scalar @$keys - 1)};
+    }
+}
+
+sub _retrieve_objects {
+    my ($self, $keys) = @_;
+    return $self->_retrieve_objects_from_cache($keys) if $self->local_caching;
+    return $self->_retrieve_objects_from_chron($keys);
+}
+
+sub _retrieve_objects_from_cache {
+    my ($self, $keys) = @_;
+    return map { $self->{$_} } @$keys;
+}
+
+sub _retrieve_objects_from_chron {
+    my ($self, $keys) = @_;
+    my @atomic_read_pairs = map { [$self->setting_namespace, $_] } @$keys;
+    return $self->chronicle_reader->mget(\@atomic_read_pairs);
+}
+
+=head2 get_history
+
+Retreives a past revision of an app config entry, where $rev is the number of revisions in the past requested.
+If the third argument is set to 1 the result of the query will be cached in Redis. This is useful if a certain
+    revision will be needed repeatedly, to avoid excessive database access. By default this argument is 0.
+All cached revisions will become stale if the key is set with a new value.
+
+Example:
+    get_history('system.email', 0); Retrieves current version
+    get_history('system.email', 1); Retreives previous revision
+    get_history('system.email', 2); Retreives version before previous
+
+=cut
+
+sub get_history {
+    my ($self, $key, $rev, $cache) = @_;
+    $cache //= 0;
+
+    die "Cannot get history of key: $key | Key must be dynamic" unless $self->_key_is_dynamic($key);
+
+    my ($curr_obj, $hist_obj) = $self->_retrieve_objects([$key, $key . '::' . $rev]);
+
+    # If no cache, or cache is stale, get from db
+    unless ($hist_obj && $hist_obj->{_local_rev} == $curr_obj->{_local_rev}) {
+        $hist_obj = $self->chronicle_reader->get_history($self->setting_namespace, $key, $rev);
+
+        $hist_obj->{_local_rev} = $curr_obj->{_local_rev} if $hist_obj;
+        $self->_store_objects(
+            {$key . '::' . $rev => $hist_obj},
+            Date::Utility->new,
+            [
+                0,         #<-- IMPORTANT: disables archiving
+                0,         #<-- IMPORTANT: supresses publication
+                604800,    #<-- ttl = 7 days so that stale histories don't stay indefinitely.
```

Comment:
A magic number hiding so far in code. Maybe start the module with something like

    use constant REDIS_HISTORY_TTL => 7 * 86400;  # 7 days

and then use that named constant here